### PR TITLE
Fix package details missing

### PR DESCRIPTION
**Pull request by johnataylor**
the dependencies collection can be null
CC @emgarten @alpaix @yishaigalatzer

### CODE REVIEW

File: src/NuGet.Core/NuGet.Protocol.Core.v3/Model/PackageSearchMetadata.cs [hunk 1]
```diff
@@ -17,7 +18,16 @@ public class PackageSearchMetadata : IPackageSearchMetadata
         public string Authors { get; private set; }
 
         [JsonProperty(PropertyName = JsonProperties.DependencyGroups, ItemConverterType = typeof(PackageDependencyGroupConverter))]
```
Comment (johnataylor) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/Model/PackageSearchMetadata.cs:20]: why are we using this JsonProperty in this one place in the code?

**Comment by alpaix** [comment]
More elegant way of doing this is to use deserialization callback:

``` C#
[OnDeserialized]
private void OnDeserializedMethod(StreamingContext context)
{
  if (DependencySets == null)
  {
    DependencySets = Enumerable.Empty<PackageDependencyGroup>();
  }
}
```

File: src/NuGet.Core/NuGet.Protocol.Core.v3/Model/PackageSearchMetadata.cs (hunk 1 above)
Comment (alpaix) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/Model/PackageSearchMetadata.cs:20]: Facilitates clear properties mapping of wire protocol to metadata object. Allows finer control over deserialization process.

**Comment by emgarten** [comment]
:shipit:

**Pull request merged**